src/App.svelte

```diff
@@ -0,0 +1,494 @@
+<script>
+// import { onMount } from "svelte";
+let value = 0;
+let counter = 1;
+let themeValue = 1;
+let storage = [0,0];
+let operator = false;
+let operatorWhich = "";
+let firstTime = false;
+let dotClicked = false;
+
+const updateTheme = (a) => {
+	themeValue = a.target.value;	
+}
+
+const number = (x) => {
+	
+	if (operator) {
+		value = x;
+	} else if(value.toString().length == 14) {
+		return;
+	} else{
+			if (dotClicked) {
+				let valueLength = value.length;
```

Comment:
You are using `value` variable as String in some places and as Number in other places. This would create unforeseen bugs in the code. Try to prevent using same variable for different types.

Reply:
Changed